packages/components/src/tools-panel/tools-panel/hook.ts

```diff
@@ -81,44 +97,55 @@ export function useToolsPanel( props ) {
 	};
 
 	// Manage and share display state of menu items representing child controls.
-	const [ menuItems, setMenuItems ] = useState( {} );
+	const [ menuItems, setMenuItems ] = useState< ToolsPanelMenuItems >( {
+		default: {},
+		optional: {},
+	} );
 
 	const getResetAllFilters = () => {
-		const filters = [];
+		const filters: Array< Function > = [];
 
-		panelItems.forEach( ( item ) => {
+		panelItems.forEach( ( item: ToolsPanelItem ) => {
```

Comment:
TypeScript already knows that `panelItems` is of type `ToolsPanelMenuItems`, and that therefore, `item` is of that same type

```suggestion
		panelItems.forEach( ( item ) => {
```